contracts/margined_vamm/src/handle.rs

```diff
@@ -132,6 +132,7 @@ pub fn swap_input(
 
     require_open(state.open)?;
     require_margin_engine(info.sender, config.margin_engine)?;
+    require_non_zero_input(quote_asset_amount)?;
```

Comment:
if you make this check before `get_input_price_with_reserves`, do you need the check [here?](https://github.com/margined-protocol/perpetuals/blob/937adde03aef7b4a110b031fd66f7c1a4474fd1b/contracts/margined_vamm/src/handle.rs#L307)